### PR TITLE
[stm] TChunk and TSchedule

### PR DESCRIPTION
**Pull request by fwbrasil**
Transactional versions for a couple of `kyo-data` APIs

### CODE REVIEW

File: kyo-stm/shared/src/main/scala/kyo/TChunk.scala [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+      * Use this only for static initialization or when you specifically need non-transactional creation. For most cases, prefer `init`.
+      *
+      * @param chunk
+      *   The initial chunk to wrap
+      * @return
+      *   A new transactional chunk containing the values, within the IO effect
+      */
+    def initNow[A](chunk: Chunk[A])(using Frame): TChunk[A] < IO = TRef.initNow(chunk)
+
+    extension [A](self: TChunk[A])
```
Comment (hearnadam) [inline kyo-stm/shared/src/main/scala/kyo/TChunk.scala:89]: Do we need a `use` method?

Comment (fwbrasil) [reply kyo-stm/shared/src/main/scala/kyo/TChunk.scala:89]: The `use` in `TMap` provides key access. Would it be indexed-based for a `Chunk`? It doesn't seem very safe.

Comment (hearnadam) [reply kyo-stm/shared/src/main/scala/kyo/TChunk.scala:89]: My suggestion, was simply `def use[B](f: Chunk[A] => B)` to enable accessing non-exposed methods.

Comment (fwbrasil) [reply kyo-stm/shared/src/main/scala/kyo/TChunk.scala:91]: Thanks for the clarification! Updated

**Comment by fwbrasil** [comment]
I'm starting an optimization round of `kyo-stm` and having this change merged will be helpful to avoid conflicts. Please feel free to provide feedback and I'll address it!

**Pull request merged**